crates/chain/src/tx_graph.rs

```diff
@@ -205,6 +205,26 @@ impl<A> TxGraph<A> {
         self.get_tx_node(txid).map(|n| n.tx)
     }
 
+    /// Get a transaction by txid. This only returns `Some` for full, unconfirmed transactions.
+    pub fn get_unconfirmed_tx<'g, C>(
```

Comment:
Try not to introduce new API if it's not needed. This method is only used externally, and as a public API, `get_chain_position` can be used instead.